Source/Provers/SMTLib/SMTLibProcess.cs

```diff
@@ -113,85 +128,60 @@ public override void Send(string cmd)
       toProver.WriteLine(cmd);
     }
 
-    internal Inspector Inspector
-    {
-      get { return inspector; }
-    }
+    internal Inspector Inspector => inspector;
 
-    public override SExpr GetProverResponse()
+    public override async Task<SExpr> GetProverResponse()
     {
-      toProver.Flush();
+      await toProver.FlushAsync();
 
-      while (true)
-      {
-        var exprs = ParseSExprs(true).ToArray();
-        Contract.Assert(exprs.Length <= 1);
-        if (exprs.Length == 0)
-        {
+      while (true) {
+        var exprs = await ParseSExprs(true).ToListAsync();
+        Contract.Assert(exprs.Count <= 1);
+        if (exprs.Count == 0) {
           return null;
         }
 
         var resp = exprs[0];
```

Comment:
The existing code ignores possible malformed output from the underlying prover, right?  I guess it's OK since it was already like that.